qfieldsync/core/cloud_api.py

```diff
@@ -733,6 +943,29 @@ def _clear_cloud_cookies(self, url: QUrl) -> None:
         for cookie in self._nam.cookieJar().cookiesForUrl(url):
             self._nam.cookieJar().deleteCookie(cookie)
 
+    def _add_csrf_cookies_to_request(self, url: QUrl, request: QNetworkRequest) -> None:
+        """
+        Adds CSRF cookie to a network request.
+        Required for some QFieldCloud requests, like POST/PUT/PATCH/DELETE.
+        """
+        cookie_jar = self._nam.cookieJar()
+        cookies = cookie_jar.cookiesForUrl(url)
+
+        csrftoken_cookie = next(
+            (cookie for cookie in cookies if cookie.name() == b"csrftoken"), None
+        )
+
+        if csrftoken_cookie:
+            request.setRawHeader(b"X-CSRFToken", csrftoken_cookie.value())
```

Comment:
Do not hardcode the header. Best case should come from QFC, worst case should be a const.